commandapi-core/src/main/java/dev/jorel/commandapi/CommandAPIHandler.java

```diff
@@ -343,15 +346,17 @@ Object parseArgument(CommandContext<Source> cmdCtx, String key, Argument value,
 	 * @param requirements An arbitrary additional check to perform on the CommandSender
 	 *                        after the permissions check
 	 */
-	Predicate<Source> generatePermissions(String commandName, CommandPermission permission,
-			Predicate<CommandSender> requirements) {
-		// If commandName was already registered, always use the first permission used
-		if (registeredPermissions.containsKey(commandName.toLowerCase())) {
-			if (!registeredPermissions.get(commandName.toLowerCase()).equals(permission)) {
-				permission = registeredPermissions.get(commandName.toLowerCase());
-			}
+	Predicate<Source> generatePermissions(String commandName, CommandPermission permission, Predicate<CommandSender> requirements, String namespace) {
+		// If namespace:commandName was already registered, always use the first permission used
+		String namespacedCommand = namespace.isEmpty()
+			? commandName.toLowerCase()
+			: namespace.toLowerCase() + ":" + commandName.toLowerCase();
+		if (registeredPermissions.containsKey(namespacedCommand)) {
+			permission = registeredPermissions.get(namespacedCommand);
 		} else {
-			registeredPermissions.put(commandName.toLowerCase(), permission);
+			registeredPermissions.put(namespacedCommand, permission);
+			// The first command to be registered determines the permission for the `commandName` version of the command
+			registeredPermissions.putIfAbsent(commandName, permission);
```

Comment:
`commandName` is case-sensitive in this map - should this have a `.toLowerCase()` here?